src/System.ComponentModel.EventBasedAsync/tests/AsyncOperationManagerTests.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using Xunit;
+
+namespace System.ComponentModel.EventBasedAsync.Tests
+{
+    public class AsyncOperationManagerTests : IDisposable
+    {
+        private readonly SynchronizationContext _originalContext;
+
+        public AsyncOperationManagerTests()
+        {
+            _originalContext = SynchronizationContext.Current;
+        }
+
+        [Fact]
+        public void SetNullSynchronizationContext()
+        {
+            var oldContext = AsyncOperationManager.SynchronizationContext;
+            Assert.NotNull(oldContext);
+
+            AsyncOperationManager.SynchronizationContext = null;
+            var newContext = AsyncOperationManager.SynchronizationContext;
+            Assert.NotNull(newContext);
+            Assert.NotSame(oldContext, newContext);
+        }
+
+        [Fact]
+        public void SetNonNullSynchronizationContext()
+        {
+            Assert.NotNull(AsyncOperationManager.SynchronizationContext);
+
+            var testContext = new SynchronizationContext();
+            AsyncOperationManager.SynchronizationContext = testContext;
+            Assert.Same(testContext, AsyncOperationManager.SynchronizationContext);
+        }
+
+        public void Dispose()
+        {
+            SynchronizationContext.SetSynchronizationContext(_originalContext);
```

Comment:
Does xunit guarantee/document that it'll run the ctor, then all of the facts, and then dispose all on the same thread and with no other facts from other tests interleaved on that thread in between the ctor and dispose?  If not, this usage of synchronization context could be problematic, and it'd be better to do the setup and teardown within each fact.